myokit/formats/diffsl/_exporter.py

```diff
@@ -0,0 +1,523 @@
+                ' units of ' + str(unit) + '.'
+            )
+
+    def _create_diffsl_variable_names(self, model):
+        """
+        Create diffsl-compatible names for all variables in the model.
+
+        The following strategy is followed:
+         - Fully qualified names are used for all variables.
+         - Variables are checked for special names, and changed if necessary.
+         - Unsupported characters like '.' and '_' are replaced.
+         - Any conflicts are resolved in a final step by appending a number.
+        """
+
+        # Detect names with special meanings
+        def special_start(name):
+            prefixes = ['dudt', 'F', 'G', 'in', 'M', 'out', 'u']
+            for prefix in prefixes:
+                if name.startswith(prefix + '_'):
+                    return True
+            return False
+
+        # Replacement characters for underscores and periods
+        uscore = 'Z'
+        period = 'Z'
+
+        # Create initial variable names
+        var_to_name = collections.OrderedDict()
+        for var in model.variables(deep=True, sort=True):
+            # Start from fully qualified name
+            name = var.qname()
+
+            # Avoid names with special meaning
+            if special_start(name):
+                name = 'var' + uscore + name
+
+            # Replace unsupported characters
+            name = name.replace('.', period).replace('_', uscore)
+
+            # Store (initial) name for var
+            var_to_name[var] = name
+
+        # Check for conflicts with known keywords
+        from . import keywords
+
+        needs_renaming = collections.OrderedDict()
+        for keyword in keywords:
+            needs_renaming[keyword] = []
+
+        pace = model.binding('pace')  # Reserve 'Vc' for pace variable
+        if pace is not None:
+            needs_renaming['Vc'] = []
+
+        # Find naming conflicts, create inverse mapping
+        name_to_var = collections.OrderedDict()
+        for var, name in var_to_name.items():
+
+            # Known conflict?
+            if name in needs_renaming.keys():
+                needs_renaming[name].append(var)
+                continue
+
+            # Test for new conflicts
+            var2 = name_to_var.get(name, None)
+            if var2 is not None:
+                needs_renaming[name] = [var2, var]
+                continue
+
+            name_to_var[name] = var
+
+        # Resolve naming conflicts
+        for name, variables in needs_renaming.items():
+            # Add a number to the end of the name, increasing it until it's
+            # unique
+            i = 1
+            root = name
+            for var in variables:
+                name = f'{root}{uscore}{i}'
+                while name in name_to_var:
+                    i += 1
+                    name = f'{root}{uscore}{i}'
+                var_to_name[var] = name
+                name_to_var[name] = var
+
+        if pace is not None:
```

Comment:
Needs documented or made more general

Reply:
Resolved by 1a212493. Pace variable keeps its original name and is no longer renamed `Vc`.